test/test_torch.py

```diff
@@ -1724,6 +1729,22 @@ def test_bool_sub(self):
                                    r"If you are trying to invert a mask, use the `\~` or `bitwise_not\(\)` operator instead.",
                                    lambda: m2 - 1)
 
+    def test_sub(self):
+        for dtype in torch.testing.get_all_dtypes():
+            m1 = torch.tensor([2.34, 4.44], dtype=dtype)
+            m2 = torch.tensor([1.23, 2.33], dtype=dtype)
+
+            if (dtype == torch.half):
+                self.assertRaises(RuntimeError, lambda: m1 - m2)
+            elif (dtype == torch.bfloat16):
+                # bfloat16 has a lower precision so we have to have a separate check for it
+                self.assertEqual(m1 - m2, torch.tensor([1.1016, 2.1094], dtype=dtype))
```

Comment:
I still don't understand why we do this instead of just assertClose or whatever with a different atol/rtol.